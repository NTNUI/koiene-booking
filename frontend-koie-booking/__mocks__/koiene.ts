import AdminBooking, { AdminBookingDictionary } from '../src/types/admin/AdminBooking';

export default function getKoieData(): AdminBookingDictionary {
  return koieData;
}

const koieData: AdminBookingDictionary = {
  ['flåkoia']: {
    name: 'Flåkoia',
    slug: 'flåkoia',
    numberOfBeds: 11,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 0,
      ['2020-09-16']: 0,
      ['2020-09-17']: 0,
      ['2020-09-18']: 11,
      ['2020-09-19']: 9,
      ['2020-09-20']: 0,
      ['2020-09-21']: 11,
      ['2020-09-22']: 11,
      ['2020-09-23']: 11,
      ['2020-09-24']: 11,
    },
  },
  ['mevasskoia']: {
    name: 'Mevasskoia',
    slug: 'mevasskoia',
    numberOfBeds: 5,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 5,
      ['2020-09-16']: 5,
      ['2020-09-17']: 5,
      ['2020-09-18']: 1,
      ['2020-09-19']: 0,
      ['2020-09-20']: 3,
      ['2020-09-21']: 5,
      ['2020-09-22']: 2,
      ['2020-09-23']: 5,
      ['2020-09-24']: 5,
    },
  },
  ['øvensenget']: {
    name: 'Øvensenget',
    slug: 'øvensenget',
    numberOfBeds: 8,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 8,
      ['2020-09-16']: 8,
      ['2020-09-17']: 8,
      ['2020-09-18']: 8,
      ['2020-09-19']: 8,
      ['2020-09-20']: 8,
      ['2020-09-21']: 8,
      ['2020-09-22']: 8,
      ['2020-09-23']: 8,
      ['2020-09-24']: 8,
    },
  },
  ['kåsen']: {
    name: 'Kåsen',
    slug: 'kåsen',
    numberOfBeds: 8,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 8,
      ['2020-09-16']: 8,
      ['2020-09-17']: 8,
      ['2020-09-18']: 8,
      ['2020-09-19']: 8,
      ['2020-09-20']: 8,
      ['2020-09-21']: 8,
      ['2020-09-22']: 8,
      ['2020-09-23']: 8,
      ['2020-09-24']: 8,
    },
  },
  ['stakkslettbua']: {
    name: 'Stakkslettbua',
    slug: 'stakkslettbua',
    numberOfBeds: 11,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 11,
      ['2020-09-16']: 11,
      ['2020-09-17']: 11,
      ['2020-09-18']: 11,
      ['2020-09-19']: 11,
      ['2020-09-20']: 11,
      ['2020-09-21']: 11,
      ['2020-09-22']: 11,
      ['2020-09-23']: 11,
      ['2020-09-24']: 11,
    },
  },
  ['holmsåkoia']: {
    name: 'Holmsåkoia',
    slug: 'holmsåkoia',
    numberOfBeds: 20,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 20,
      ['2020-09-16']: 20,
      ['2020-09-17']: 20,
      ['2020-09-18']: 20,
      ['2020-09-19']: 20,
      ['2020-09-20']: 20,
      ['2020-09-21']: 20,
      ['2020-09-22']: 20,
      ['2020-09-23']: 20,
      ['2020-09-24']: 20,
    },
  },
  ['sonvasskoia']: {
    name: 'Sonvasskoia',
    slug: 'sonvasskoia',
    numberOfBeds: 8,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 8,
      ['2020-09-16']: 8,
      ['2020-09-17']: 8,
      ['2020-09-18']: 8,
      ['2020-09-19']: 8,
      ['2020-09-20']: 8,
      ['2020-09-21']: 8,
      ['2020-09-22']: 8,
      ['2020-09-23']: 8,
      ['2020-09-24']: 8,
    },
  },
  ['nicokoia']: {
    name: 'Nicokoia',
    slug: 'nicokoia',
    numberOfBeds: 8,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 8,
      ['2020-09-16']: 8,
      ['2020-09-17']: 8,
      ['2020-09-18']: 8,
      ['2020-09-19']: 8,
      ['2020-09-20']: 8,
      ['2020-09-21']: 8,
      ['2020-09-22']: 8,
      ['2020-09-23']: 8,
      ['2020-09-24']: 8,
    },
  },
  ['telin']: {
    name: 'Telin',
    slug: 'telin',
    numberOfBeds: 9,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 9,
      ['2020-09-16']: 9,
      ['2020-09-17']: 9,
      ['2020-09-18']: 9,
      ['2020-09-19']: 9,
      ['2020-09-20']: 9,
      ['2020-09-21']: 9,
      ['2020-09-22']: 9,
      ['2020-09-23']: 9,
      ['2020-09-24']: 9,
    },
  },
  ['lyngli']: {
    name: 'Lyngli',
    slug: 'lyngli',
    numberOfBeds: 13,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 13,
      ['2020-09-16']: 13,
      ['2020-09-17']: 13,
      ['2020-09-18']: 13,
      ['2020-09-19']: 13,
      ['2020-09-20']: 13,
      ['2020-09-21']: 13,
      ['2020-09-22']: 13,
      ['2020-09-23']: 13,
      ['2020-09-24']: 13,
    },
  },
  ['kråklikåten']: {
    name: 'Kråklikåten',
    slug: 'kråklikåten',
    numberOfBeds: 4,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 4,
      ['2020-09-16']: 4,
      ['2020-09-17']: 4,
      ['2020-09-18']: 4,
      ['2020-09-19']: 4,
      ['2020-09-20']: 4,
      ['2020-09-21']: 4,
      ['2020-09-22']: 4,
      ['2020-09-23']: 4,
      ['2020-09-24']: 4,
    },
  },
  ['heinfjordstua']: {
    name: 'Heinfjordstua',
    slug: 'heinfjordstua',
    numberOfBeds: 25,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 25,
      ['2020-09-16']: 25,
      ['2020-09-17']: 25,
      ['2020-09-18']: 25,
      ['2020-09-19']: 25,
      ['2020-09-20']: 25,
      ['2020-09-21']: 25,
      ['2020-09-22']: 25,
      ['2020-09-23']: 25,
      ['2020-09-24']: 25,
    },
  },
  ['høgnabu']: {
    name: 'Høgnabu',
    slug: 'høgnabu',
    numberOfBeds: 6,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 6,
      ['2020-09-16']: 6,
      ['2020-09-17']: 6,
      ['2020-09-18']: 6,
      ['2020-09-19']: 6,
      ['2020-09-20']: 6,
      ['2020-09-21']: 6,
      ['2020-09-22']: 6,
      ['2020-09-23']: 6,
      ['2020-09-24']: 6,
    },
  },
  ['stabburet']: {
    name: 'Stabburet',
    slug: 'stabburet',
    numberOfBeds: 2,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 2,
      ['2020-09-16']: 2,
      ['2020-09-17']: 2,
      ['2020-09-18']: 2,
      ['2020-09-19']: 2,
      ['2020-09-20']: 2,
      ['2020-09-21']: 2,
      ['2020-09-22']: 2,
      ['2020-09-23']: 2,
      ['2020-09-24']: 2,
    },
  },
  ['vekvessætra']: {
    name: 'Vekvessætra',
    slug: 'vekvessætra',
    numberOfBeds: 20,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 20,
      ['2020-09-16']: 20,
      ['2020-09-17']: 20,
      ['2020-09-18']: 20,
      ['2020-09-19']: 20,
      ['2020-09-20']: 20,
      ['2020-09-21']: 20,
      ['2020-09-22']: 20,
      ['2020-09-23']: 20,
      ['2020-09-24']: 20,
    },
  },
  ['fosenkoia']: {
    name: 'Fosenkoia',
    slug: 'fosenkoia',
    numberOfBeds: 10,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 10,
      ['2020-09-16']: 10,
      ['2020-09-17']: 10,
      ['2020-09-18']: 10,
      ['2020-09-19']: 10,
      ['2020-09-20']: 10,
      ['2020-09-21']: 10,
      ['2020-09-22']: 10,
      ['2020-09-23']: 10,
      ['2020-09-24']: 10,
    },
  },
  ['tågåbu']: {
    name: 'Tågåbu',
    slug: 'tågåbu',
    numberOfBeds: 6,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 6,
      ['2020-09-16']: 6,
      ['2020-09-17']: 6,
      ['2020-09-18']: 6,
      ['2020-09-19']: 6,
      ['2020-09-20']: 6,
      ['2020-09-21']: 6,
      ['2020-09-22']: 6,
      ['2020-09-23']: 6,
      ['2020-09-24']: 6,
    },
  },
  ['lynhøgen']: {
    name: 'Lynhøgen',
    slug: 'lynhøgen',
    numberOfBeds: 4,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 4,
      ['2020-09-16']: 4,
      ['2020-09-17']: 4,
      ['2020-09-18']: 4,
      ['2020-09-19']: 4,
      ['2020-09-20']: 4,
      ['2020-09-21']: 4,
      ['2020-09-22']: 4,
      ['2020-09-23']: 4,
      ['2020-09-24']: 4,
    },
  },
  ['selbukåten']: {
    name: 'Selbukåten',
    slug: 'selbukåten',
    numberOfBeds: 2,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 2,
      ['2020-09-16']: 2,
      ['2020-09-17']: 2,
      ['2020-09-18']: 2,
      ['2020-09-19']: 2,
      ['2020-09-20']: 2,
      ['2020-09-21']: 2,
      ['2020-09-22']: 2,
      ['2020-09-23']: 2,
      ['2020-09-24']: 2,
    },
  },
  ['holvassgamma']: {
    name: 'Holvassgamma',
    slug: 'holvassgamma',
    numberOfBeds: 8,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 8,
      ['2020-09-16']: 8,
      ['2020-09-17']: 8,
      ['2020-09-18']: 8,
      ['2020-09-19']: 8,
      ['2020-09-20']: 8,
      ['2020-09-21']: 8,
      ['2020-09-22']: 8,
      ['2020-09-23']: 8,
      ['2020-09-24']: 8,
    },
  },
  ['kamtjønnkoia']: {
    name: 'Kamtjønnkoia',
    slug: 'kamtjønnkoia',
    numberOfBeds: 6,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 6,
      ['2020-09-16']: 6,
      ['2020-09-17']: 6,
      ['2020-09-18']: 6,
      ['2020-09-19']: 6,
      ['2020-09-20']: 6,
      ['2020-09-21']: 6,
      ['2020-09-22']: 6,
      ['2020-09-23']: 6,
      ['2020-09-24']: 6,
    },
  },
  ['rindalsløa']: {
    name: 'Rindalsløa',
    slug: 'rindalsløa',
    numberOfBeds: 4,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 4,
      ['2020-09-16']: 4,
      ['2020-09-17']: 4,
      ['2020-09-18']: 4,
      ['2020-09-19']: 4,
      ['2020-09-20']: 4,
      ['2020-09-21']: 4,
      ['2020-09-22']: 4,
      ['2020-09-23']: 4,
      ['2020-09-24']: 4,
    },
  },
  ['iglbu']: {
    name: 'Iglbu',
    slug: 'iglbu',
    numberOfBeds: 10,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 10,
      ['2020-09-16']: 10,
      ['2020-09-17']: 10,
      ['2020-09-18']: 10,
      ['2020-09-19']: 10,
      ['2020-09-20']: 10,
      ['2020-09-21']: 10,
      ['2020-09-22']: 10,
      ['2020-09-23']: 10,
      ['2020-09-24']: 10,
    },
  },
  ['mortenskåten']: {
    name: 'Mortenskåten',
    slug: 'mortenskåten',
    numberOfBeds: 2,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 2,
      ['2020-09-16']: 2,
      ['2020-09-17']: 2,
      ['2020-09-18']: 2,
      ['2020-09-19']: 2,
      ['2020-09-20']: 2,
      ['2020-09-21']: 2,
      ['2020-09-22']: 2,
      ['2020-09-23']: 2,
      ['2020-09-24']: 2,
    },
  },
};
