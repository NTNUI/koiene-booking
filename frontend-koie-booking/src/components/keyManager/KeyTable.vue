<template>
  <v-row style="height: 60%; background-color: black">
    <v-row justify="center">
      <v-col>
        {{ title }}
      </v-col>
      <v-col>
        {{ description }}
      </v-col>
    </v-row>
    <v-data-table
      style="width: 100%"
      :headers="headers"
      :items="items"
      :items-per-page="24"
      :hide-default-footer="true"
      :loading="loading"
      loading-text="Henter inn status for nøkler"
      no-data-text="Det skjedde noe feil da vi prøvde å hente status for nøkler"
    >
      <template slot="item" slot-scope="row">
        <slot name="row" :row="row" />
      </template>
    </v-data-table>
  </v-row>
</template>

<script lang="ts">
import Vue from 'vue';

import keyTableHeaders from '@/components/keyManager/keyTableHeaders';
import KeyDetail from '@/types/keyManager/KeyDetail';

export default Vue.extend({
  name: 'KeyTable',
  props: {
    title: {
      type: String,
      default: '',
    },
    description: {
      type: String,
      default: '',
    },
    items: {
      type: Object as () => KeyDetail[],
      default() {
        return [];
      },
    },
    loading: {
      type: Boolean,
      default: false,
    },
  },
  data() {
    return {
      headers: keyTableHeaders,
    };
  },
});
</script>

<style scoped></style>
