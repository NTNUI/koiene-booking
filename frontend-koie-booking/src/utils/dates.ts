import dayjs from 'dayjs';
require('dayjs/locale/nb');
dayjs.locale('nb');

export function getDateString(baseDate?: string, addDays?: number): string {
  let resultString: string;
  if (baseDate && addDays) {
    resultString = new Date(new Date(baseDate).getTime() + addDays * 24 * 60 * 60 * 1000).toISOString().substr(0, 10);
  } else if (baseDate) {
    resultString = new Date(baseDate).toISOString().substr(0, 10);
  } else if (addDays) {
    resultString = new Date(new Date().getTime() + addDays * 24 * 60 * 60 * 1000).toISOString().substr(0, 10);
  } else {
    resultString = new Date().toISOString().substr(0, 10);
  }
  return resultString;
}

export function getDate(baseDate?: string, addDays?: number): Date {
  let resultDate: Date;
  if (baseDate && addDays) {
    resultDate = new Date(new Date(baseDate).getTime() + addDays * 24 * 60 * 60 * 1000);
  } else if (baseDate) {
    resultDate = new Date(baseDate);
  } else if (addDays) {
    resultDate = new Date(new Date().getTime() + addDays * 24 * 60 * 60 * 1000);
  } else {
    resultDate = new Date();
  }
  return resultDate;
}

export function addToDate(startDate: string, howMany: number, timeUnit: dayjs.OpUnitType): string {
  const date = dayjs(startDate + 'T00:00:00.000Z');
  return date.add(howMany, timeUnit).format('YYYY-MM-DD');
}

export function formatDate(dateISO: string, formatString: string): string {
  const date = dayjs(dateISO + 'T00:00:00.000Z');
  return date.format(formatString);
}
