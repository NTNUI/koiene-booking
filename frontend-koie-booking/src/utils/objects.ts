export const clone = (object: any) => Object.assign({}, object);

export const renameKey = (object: any, key: string, newKey: string): any => {
  const value = object[key];
  delete object[key];
  object[newKey] = value;
  return object;
};

export const sortObjectByKey = (unordered: any) => {
  let ordered: any;
  ordered = {};
  Object.keys(unordered)
    .sort()
    .forEach(function(key) {
      ordered[key] = unordered[key];
    });
  return ordered;
};
