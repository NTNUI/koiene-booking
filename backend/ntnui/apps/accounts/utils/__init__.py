from accounts.utils.mail_utils import send_gsuite_confirmation_mail
