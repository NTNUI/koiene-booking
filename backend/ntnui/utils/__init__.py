from ntnui.utils.pagination import PageSize7, PageSize20, PageSize100
