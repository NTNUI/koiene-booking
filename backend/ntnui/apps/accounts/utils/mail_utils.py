from django.core.mail import send_mail
from django.template.loader import render_to_string

from ntnui.enums import MembershipType


def new_membership(membership):
    """ Sends confirmation email to users and guests who signs up for events and sub-events. """

    # Mail header.
    sender = "noreply@mg.ntnui.no"
    receiver = [membership.member.email]
    subject = "Welcome"

    context = {"user": membership.member, "group": membership.group}

    # Mail body.
    plain_message = render_to_string("emails/new_membership.txt", context)
    html_message = render_to_string("emails/new_membership.html", context)

    send_mail(subject, plain_message, sender, receiver, html_message=html_message)


def send_gsuite_confirmation_mail(user, contact_email, password):
    # Mail header.
    sender = "noreply@mg.ntnui.no"
    receiver = [user.email]
    subject = "Opprettelse av ny NTNUI-bruker"

    context = {
        "first_name": user.first_name,
        "email": user.email,
        "contact_email": contact_email,
        "password": password,
    }

    # Mail body.
    plain_message = render_to_string("emails/new_gsuite_account.txt", context)
    html_message = render_to_string("emails/new_gsuite_account.html", context)

    send_mail(subject, plain_message, sender, receiver, html_message=html_message)


def send_welcome_mail_to_leader(user, group):
    """ Send a mail to a new leader """
    sender = "noreply@mg.ntnui.no"
    receiver = [user.email]
    subject = f"Velkommen som ny leder i {group}"

    context = {
        "first_name": user.first_name,
        "email": user.get_contact_email(),
        "group": group.name,
    }

    plain_message = render_to_string(
        template_name="emails/new_leader_membership.txt", context=context
    )
    html_message = render_to_string(
        template_name="emails/new_leader_membership.html", context=context
    )

    send_mail(
        subject=subject,
        message=plain_message,
        from_email=sender,
        recipient_list=receiver,
        html_message=html_message,
    )


def send_welcome_mail_to_cashier(user, group):
    """Send a mail to the new cashier"""
    sender = "noreply@mg.ntnui.no"
    receiver = [user.email]
    subject = f"Velkommen som ny kasserer i {group}"

    context = {
        "first_name": user.first_name,
        "email": user.get_contact_email(),
        "group": group.name,
    }

    plain_message = render_to_string(
        template_name="emails/new_cashier_membership.txt", context=context
    )
    html_message = render_to_string(
        template_name="emails/new_cashier_membership.html", context=context
    )

    send_mail(
        subject=subject,
        message=plain_message,
        from_email=sender,
        recipient_list=receiver,
        html_message=html_message,
    )


def send_welcome_mail_to_board_member(user, group):
    """Send a mail to the new cashier"""
    sender = "noreply@mg.ntnui.no"
    receiver = [user.email]
    subject = f"Velkommen som ny tillitsvalgt i {group}"

    context = {
        "first_name": user.first_name,
        "email": user.get_contact_email(),
        "group": group.name,
    }

    plain_message = render_to_string(
        template_name="emails/new_board_membership.txt", context=context
    )
    html_message = render_to_string(
        template_name="emails/new_board_membership.html", context=context
    )

    send_mail(
        subject=subject,
        message=plain_message,
        from_email=sender,
        recipient_list=receiver,
        html_message=html_message,
    )


def send_welcome_mail(membership):
    """ Delegate welcome mail to different board members """
    send_welcome_mail_to_board_member(user=membership.member, group=membership.group)
    if membership.type == MembershipType.leader:
        send_welcome_mail_to_leader(user=membership.member, group=membership.group)

    if membership.type == MembershipType.cashier:
        send_welcome_mail_to_cashier(user=membership.member, group=membership.group)
