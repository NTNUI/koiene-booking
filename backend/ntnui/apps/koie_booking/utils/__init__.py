from koie_booking.utils.mail_utils import send_confirmation_mail, send_koie_information_mail
from koie_booking.utils.date_utils import get_daterange
