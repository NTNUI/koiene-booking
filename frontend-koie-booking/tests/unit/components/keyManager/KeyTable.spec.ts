import Vue from 'vue';
import Vuetify from 'vuetify';
import { createShallowWrapper } from '../../utils';
import KeyTable from '@/components/keyManager/KeyTable.vue';
import dayjs from 'dayjs';
import { Wrapper } from '@vue/test-utils';
import { getKeyPickUps } from '../../__mocks__/keys';

Vue.use(Vuetify);

describe('Component KeyTable', () => {
  it('Matches snapshot', () => {
    const wrapperOptions = {
      propsData: {
        title: 'Test',
        description: 'Test table',
        items: getKeyPickUps(),
        loading: false,
      },
    };
    const wrapper = createShallowWrapper(KeyTable, wrapperOptions) as Wrapper<KeyTable>;
    expect(wrapper).toMatchSnapshot();
  });
  it('Defaults to empty array when no items are provided', () => {
    const wrapperOptions = {
      propsData: {
        title: 'Test',
        description: 'Test table',
        loading: false,
      },
    };
    const wrapper = createShallowWrapper(KeyTable, wrapperOptions);
    expect(wrapper.vm.items).toEqual([]);
  });
});
