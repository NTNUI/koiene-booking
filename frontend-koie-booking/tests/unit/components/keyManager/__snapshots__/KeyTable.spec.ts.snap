// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component KeyTable Matches snapshot 1`] = `
<div id="dataTable">
  <v-row-stub tag="div" justify="center"><span style="font-weight: bold;">Test</span>
    <v-spacer-stub></v-spacer-stub> <span>Test table</span>
  </v-row-stub>
  <v-data-table-stub items="[object Object],[object Object],[object Object]" options="[object Object]" sortby="" sortdesc="" customsort="function sortItems(items, sortBy, sortDesc, locale, customSorters) {
  if (sortBy === null || !sortBy.length) return items;
  var stringCollator = new Intl.Collator(locale, {
    sensitivity: 'accent',
    usage: 'sort'
  });
  return items.sort(function (a, b) {
    var _a, _b;

    for (var i = 0; i < sortBy.length; i++) {
      var sortKey = sortBy[i];
      var sortA = getObjectValueByPath(a, sortKey);
      var sortB = getObjectValueByPath(b, sortKey);

      if (sortDesc[i]) {
        _a = __read([sortB, sortA], 2), sortA = _a[0], sortB = _a[1];
      }

      if (customSorters &amp;&amp; customSorters[sortKey]) {
        var customResult = customSorters[sortKey](sortA, sortB);
        if (!customResult) continue;
        return customResult;
      } // Check if both cannot be evaluated


      if (sortA === null &amp;&amp; sortB === null) {
        continue;
      }

      _b = __read([sortA, sortB].map(function (s) {
        return (s || '').toString().toLocaleLowerCase();
      }), 2), sortA = _b[0], sortB = _b[1];

      if (sortA !== sortB) {
        if (!isNaN(sortA) &amp;&amp; !isNaN(sortB)) return Number(sortA) - Number(sortB);
        return stringCollator.compare(sortA, sortB);
      }
    }

    return 0;
  });
}" page="1" itemsperpage="10" groupby="" groupdesc="" customgroup="function groupItems(items, groupBy, groupDesc) {
  var key = groupBy[0];
  var groups = [];
  var current = null;

  for (var i = 0; i < items.length; i++) {
    var item = items[i];
    var val = getObjectValueByPath(item, key);

    if (current !== val) {
      current = val;
      groups.push({
        name: val,
        items: []
      });
    }

    groups[groups.length - 1].items.push(item);
  }

  return groups;
}" locale="en-US" search="" customfilter="function defaultFilter(value, search, item) {
  return value != null &amp;&amp; search != null &amp;&amp; typeof value !== 'boolean' &amp;&amp; value.toString().toLocaleLowerCase().indexOf(search.toLocaleLowerCase()) !== -1;
}" serveritemslength="-1" itemkey="id" value="" expanded="" mobilebreakpoint="600" noresultstext="$vuetify.dataIterator.noResultsText" nodatatext="Det skjedde noe feil da vi prøvde å hente status for nøkler" loadingtext="Henter inn status for nøkler" selectablekey="isSelectable" headers="[object Object],[object Object],[object Object],[object Object],[object Object]" height="100%" expandicon="$expand" style="width: 100%; background-color: rgba(0, 0, 0, 0); height: 40vh;"></v-data-table-stub>
</div>
`;
